core/state_transition.go

```diff
@@ -300,10 +302,10 @@ func (st *StateTransition) preCheck() error {
 	}
 	if !msg.SkipFromEOACheck {
 		// Make sure the sender is an EOA
-		codeHash := st.state.GetCodeHash(msg.From)
-		if codeHash != (common.Hash{}) && codeHash != types.EmptyCodeHash {
+		code := st.state.GetCode(msg.From)
+		if 0 < len(code) && !bytes.HasPrefix(code, []byte{0xef, 0x01, 0x00}) {
```

Comment:
What is this? Previously, we entered here if codehash told is it was "non-empty-code". 
Now, we enter here if length of code is `>0`, ok, but why check against `ef, 01, 00`? 
And that's not even the eof-prefix+version, which is `ef, 00,01`. 

Also please don't do `if 0 < foo`, do `if foo > 0`. We're not testing the `0`, we're testing the `foo`.

Reply:
I have no idea where that code came from, I am very certain that I didn't write it^^

Reply:
I think its fine to just revert back to the codelength check